src/node/eth.rs

```diff
@@ -1390,28 +1390,35 @@ impl<S: ForkSource + std::fmt::Debug + Clone + Send + Sync + 'static> EthTestNod
         &self,
         tx: zksync_types::transaction_request::CallRequest,
     ) -> jsonrpc_core::BoxFuture<jsonrpc_core::Result<zksync_basic_types::H256>> {
-        let chain_id = match self.get_inner().read() {
-            Ok(reader) => reader.fork_storage.chain_id,
+        let (chain_id, l1_gas_price) = match self.get_inner().read() {
+            Ok(reader) => (
+                reader.fork_storage.chain_id,
+                reader.fee_input_provider.l1_gas_price,
+            ),
             Err(_) => {
-                return futures::future::err(into_jsrpc_error(Web3Error::InternalError(
-                    anyhow::Error::msg("Failed to acquire read lock for chain ID retrieval."),
-                )))
+                return futures::future::err(into_jsrpc_error_message(
+                    "Failed to acquire read lock for chain ID retrieval.".to_string(),
+                ))
                 .boxed()
             }
         };
 
         let mut tx_req = TransactionRequest::from(tx.clone());
+        // Users might expect a "sensible default"
+        if tx.gas.is_none() {
```

Comment:
I'm concerned about this... If someone is not providing gas, doesn't that mean we should be getting the gas limit via gas estimation logic?

I'm worried that if we always default to the max limit, it's not truly reflective of what happens for mainnet/testnet

Reply:
As `send_transaction` is only used after account impersonation, gas estimation shouldn't matter (IMHO). This should be okay for now. We can always revisit if something comes up